src/js/options.js

```diff
@@ -1,39 +1,20 @@
 
-function requestPermissions() {
+async function requestPermissions() {
   const checkbox = document.querySelector("#bookmarksPermissions");
   if (checkbox.checked) {
-    browser.permissions.request({permissions: ["bookmarks"]}).
-    then((response) => {
-      if (response) {
-        browser.runtime.sendMessage({ method: "resetBookmarksContext" });
-      } else {
-        checkbox.checked = false;
-      }
-    }).
-    catch((err) => {
-      return err.message;
-    });
+    const granted = await browser.permissions.request({permissions: ["bookmarks"]});
+    if (!granted) { checkbox.checked = false; }
```

Comment:
Not sure what happened, but the `return;` disappeared here.

It should be

```javascript
if (!granted) {
  checkbox.checked = false;
  return;
}
```

Reply:
There you go!